qiime2/core/cache.py

```diff
@@ -207,6 +243,18 @@ def __exit__(self, *args):
         """
         _CACHE.cache = None
 
+    def __getstate__(self):
+        """We don't want to pickle any of the thread stuff because it won't
+        rehydrate, and we don't care about it
+        """
+        threadless_dict = self.__dict__.copy()
+
+        del threadless_dict['_thread_is_done']
```

Comment:
We should check if our cache has a `._thread` at all before deleting, as we would get a keyerror otherwise.

i.e. a un-pickled cache is pickled again, or if the above change is made and not all caches have the silly thread watcher

Reply:
`hasattr()` is the easy way to do this